ompi/mca/part/persist/part_persist_request.h

```diff
@@ -40,7 +41,7 @@ struct ompi_mca_persist_setup_t {
    int start_tag;
    int setup_tag;
    size_t num_parts;
-   size_t dt_size;
+   ssize_t dt_size;
```

Comment:
How would you create a datatype with negative size or a message with less than 0 bytes ?

Reply:
Fair enough question, which is one of the reasons I raised this PR as draft.  Without these two types as `ssize_t` in this component, I get several of these types of warnings:

```
  CC       part_persist.lo
In file included from ../../../../../ompi/mca/part/persist/part_persist.c:27:
../../../../../ompi/mca/part/persist/part_persist.h: In function 'mca_part_persist_progress':
../../../../../ompi/mca/part/persist/part_persist.h:275:42: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int32_t' {aka 'int'} [-Wsign-compare]
  275 |                     if(req->real_dt_size == dt_size) {
      |                                          ^~
```

@mdosanjh Can you comment here?

Reply:
I think the best fix is to convert the locally defined datatype size variables to size_t rather than making the these variables signed? It looks a few of the local variables were erroneously defined as ints and that's the source of the warnings.

Reply:
@mdosanjh Ok, thanks.  Could you suggest a patch?  I can replace the commit here on the PR with it.

Reply:
Patch was exchanged via email; I updated this PR to include @mdosanjh's patch (wholly replacing my own).